driver/server/grpc.go

```diff
@@ -17,46 +18,22 @@ import (
 )
 
 // NewGRPCServer creates a gRPC server.
-func NewGRPCServer(drv driver.DriverModule, opts ...grpc.ServerOption) *GRPCServer {
-	return &GRPCServer{drv: drv, Options: opts}
-}
-
-// GRPCServer is a common implementation of a gRPC server.
-type GRPCServer struct {
-	// Options list of grpc.ServerOption's.
-	Options []grpc.ServerOption
-
-	drv driver.DriverModule
-	*grpc.Server
-}
-
-// Serve accepts incoming connections on the listener lis, creating a new
-// ServerTransport and service goroutine for each.
-func (s *GRPCServer) Serve(listener net.Listener) error {
-	if err := s.initialize(); err != nil {
-		return err
-	}
-
-	defer func() {
-		logrus.Infof("grpc server ready")
-	}()
-
-	return s.Server.Serve(listener)
-}
-
-func (s *GRPCServer) initialize() error {
-	s.Server = grpc.NewServer(s.Options...)
-
-	logrus.Debugf("registering grpc service")
+func NewGRPCServer(drv driver.DriverModule, opts ...grpc.ServerOption) *grpc.Server {
```

Comment:
Is there some reason we're hard-coding these options into the constructor, rather than passing them in from main? (I assume there is a reason, but a comment might be helpful here)

Reply:
Added a comment

Reply:
On API level it makes sense to pass in all grpc options though the constructor, so a client (in this case driver Server) can choose e.g to disable tracing for some production deployment based on CLI.

Reply:
To disable tracing, you just don't pass any tracing configuration in env :)

Also, tracing does not incur the performance cost you think it does. It may apply to 1/1000 requests, for example.

So I think it's safe to integrate it by default. Because if you forget to enable it, bblfshd will not propagate trace info to the driver, for example.

Reply:
Ok, but it isn't just about cost—this is an API and if the caller is heap profiling (for example), even a 0.1% sampling rate is enough to mess up the results. 

If the only callers of this function are "us" (i.e., clients we maintain), it seem like we could pretty easily make sure they all pass the tracing options (we could even make a slice of them as a variable in the package, or a helper to populate them among other options). If there _are_ other clients, maybe we can separate these two concerns:

```
// NewGRPCServer constructs a GRPC endpoint that dispatches to the specified
// driver module for …
func NewGRPCServer(…) *grpc.Server { … }

// NewTracedGRPCServer constructs a GRPC endpoint with tracing enabled that …
func NewTracedGRPCServer(…) *grpc.Server { … return NewGRPCServer(mod, combinedOpts...) }
```

or words to that effect. I realize it seems like extra work now, but these corner cases pile up.

In any case: We don't have to do all of this right now, but if this constructor is going to have default options baked in, I'd like to see that reflected in the doc comment since it's part of the API for this package.